Foundation/NSOperation.swift

```diff
@@ -9,6 +9,12 @@
 
 #if DEPLOYMENT_ENABLE_LIBDISPATCH
 import Dispatch
+#if os(Linux)
+import CoreFoundation
+private func pthread_main_np() -> Int32 {
+    return _CFIsMainThread()
+}
+#endif
```

Comment:
Here I would turn the boolean result of _CFIsMainThread into the Int32 expected by the pthread_main_np stub.